src/components/dialogs/input-hooks.js

```diff
@@ -442,51 +476,131 @@ export const useAutocompleteField = ({
         }
     }, [selectedValue]);
 
+    useEffect(() => {
+        if (!values || typeof values === 'function') {
+            setIsLoading(false);
+            return;
+        }
+        if (typeof values?.then === 'function') {
+            setIsLoading(true);
+        }
+        const valuePromise = Promise.resolve(values);
+        valuePromise.then((vals) => {
+            setPresentedOptions(vals);
+            setIsLoading(false);
+            if (values?.length === 0) setExpanded(false);
+        });
+    }, [values]);
+
+    const handleForcedSearch = useCallback(
+        (term) => {
+            if (!onSearchTermChange) return;
+            setIsLoading(true);
+            setExpanded(true);
+            onSearchTermChange(term, true);
+        },
+        [onSearchTermChange]
+    );
+
+    const myOnOpen = useCallback(() => {
+        setExpanded(true);
+
+        if (!onSearchTermChange) return;
+        if (isWorthLoading(userStr, presentedOptions, userStr, 0)) {
+            setIsLoading(true);
+            onSearchTermChange(userStr, false);
+        }
+    }, [presentedOptions, userStr, onSearchTermChange]);
+
+    const handleSearchTermChange = useCallback(
+        (term) => {
+            if (!onSearchTermChange) return;
+
+            const min = minCharsBeforeSearch;
+
+            setUserStr((old) => {
+                if (isWorthLoading(term, values, old, min)) {
+                    setIsLoading(true);
+                    setExpanded(true);
+                    onSearchTermChange(term, false);
+                }
+                return term;
+            });
+        },
+        [values, minCharsBeforeSearch, onSearchTermChange]
+    );
+
     const field = useMemo(() => {
+        const handleKeyDown = (e) => {
+            if (e.ctrlKey && e.code === 'Space') {
+                handleForcedSearch(userStr);
+            }
+        };
+
+        const optionEqualsToValue = (option, input) =>
```

Comment:
function returning Boolean should start with "is"

Reply:
This is a convention for boolean values too, and should not be applied blindly, though it could have been applied profitably to what is named freeSolo not far from here

Reply:
I just thought it would improve readability here